src/Cli/Microsoft.DotNet.Cli.Utils/ForwardingAppImplementation.cs

```diff
@@ -96,7 +97,19 @@ public ForwardingAppImplementation WithEnvironmentVariable(string name, string v
         private string GetHostExeName()
         {
             // Should instead make this a full path to dotnet
-            return Environment.ProcessPath;
+            string processPath = Environment.ProcessPath;
+            string hostName = "dotnet" + Constants.ExeSuffix;
+
+            if (!processPath.EndsWith(hostName, StringComparison.OrdinalIgnoreCase))
+            {
+                var dotnetHostPath = Environment.GetEnvironmentVariable("DOTNET_HOST_PATH") ?? Environment.GetEnvironmentVariable("DOTNET_ROOT");
+                if (!string.IsNullOrEmpty(dotnetHostPath))
+                {
+                    processPath = dotnetHostPath+ Path.DirectorySeparatorChar + hostName;
+                }
+            }
+
+            return processPath;
         }
```

Comment:
```suggestion
        private string GetHostExeName() => new Muxer().MuxerPath;
```
(`MuxerPath` property can be static to avoid new'ing the instance)